packages/cli-repl/src/config-directory.ts

```diff
@@ -0,0 +1,76 @@
+import { promises as fs } from 'fs';
+import path from 'path';
+import { EventEmitter } from 'events';
+
+export class ConfigAndLogDirectoryManager<Config> extends EventEmitter {
```

Comment:
What about decoupling it even more, moving the accent to the fact that there is a Config and some Logs: 

``` js
class ShellHomeDir {
    ensure(...) {} ...
    path() {}
}
```

``` js
class ConfigManager<Config> {
     constructor(shellHomeDir) {}
 }
```

``` js
class LogStuff {
     constructor(shellHomeDir) {}
 }
```

Reply:
I mean, the log stuff is basically just a single call to `path` :woman_shrugging: In that case it probably makes sense to drop the `Log` part from the name, even though most files in that path will be log files

Reply:
Well for the purpose of making it more testable i think we nailed it by doing what you've done. 

From the perspective of isolating concerns it still feels a bit mixed. I mean there are still some fragments in cli repl, something here is both about the config and logs as well. 

Don't know, I have this feeling we could go one step further and completely isolate CliRepl from any knowledge about directories, how things are stored etc. 

Leaving the CliRepl code with only knowledge that's needed for it to operate: like the CliRepl only needs to know if the telemetry is enabled to show the greeting message, is not necessary to also know how that's stored.

Reply:
@mcasimir Yeah, I just changed this to basically adopt your suggestion here :slightly_smiling_face: It makes sense to separate the directory management from config management, I just left out a specific class for logging facilities because that would basically be empty for now